pkg/bff/collaborators.go

```diff
@@ -65,74 +64,60 @@ func (s *Server) AddCollaborator(c *gin.Context) {
 		return
 	}
 
-	// Make sure the collaborator is valid for storage
-	if err = collaborator.Validate(); err != nil {
-		log.Warn().Err(err).Msg("invalid collaborator in request")
-		c.JSON(http.StatusBadRequest, api.ErrorResponse(err))
-		return
-	}
-
-	// Limit the number of collaborators
-	if len(org.Collaborators) >= MaxCollaborators {
-		log.Warn().Int("current", len(org.Collaborators)).Int("maximum", MaxCollaborators).Msg("maximum number of collaborators reached")
-		c.JSON(http.StatusForbidden, api.ErrorResponse("maximum number of collaborators reached"))
-		return
-	}
-
-	// Don't overwrite an existing collaborator
-	id := collaborator.Key()
-	if _, ok := org.Collaborators[id]; ok {
-		log.Warn().Str("collabID", id).Msg("collaborator already exists")
-		c.JSON(http.StatusConflict, api.ErrorResponse("collaborator already exists"))
-		return
-	}
-
-	// Make sure the record has a created timestamp
-	if collaborator.CreatedAt == "" {
-		collaborator.CreatedAt = time.Now().UTC().Format(time.RFC3339Nano)
-	}
+	// Collaborator invites are valid for seven days
+	collaborator.ExpiresAt = time.Now().AddDate(0, 0, 7).Format(time.RFC3339Nano)
 
 	// Add the collaborator to the organization
-	if org.Collaborators == nil {
-		org.Collaborators = make(map[string]*models.Collaborator)
+	if err = org.AddCollaborator(collaborator); err != nil {
+		switch {
+		case errors.Is(err, models.ErrInvalidCollaborator):
+			log.Warn().Err(err).Msg("invalid collaborator")
+			c.JSON(http.StatusBadRequest, api.ErrorResponse(err))
+		case errors.Is(err, models.ErrCollaboratorExists):
+			log.Warn().Err(err).Str("email", collaborator.Email).Msg("collaborator already exists")
+			c.JSON(http.StatusConflict, api.ErrorResponse(err))
+		case errors.Is(err, models.ErrMaxCollaborators):
+			log.Warn().Err(err).Int("maximum", models.MaxCollaborators).Msg("maximum number of collaborators reached")
+			c.JSON(http.StatusForbidden, api.ErrorResponse(err))
```

Comment:
Will this error still happen? It looks like we have removed the `MaxCollaborators` const.

Reply:
Nevermind! Looks like it's just moved to `pkg/bff/models/v1/organization.go`